sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/DocGenerationHelper.java

```diff
@@ -0,0 +1,85 @@
+package de.symeda.sormas.backend.docgeneration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import de.symeda.sormas.api.ReferenceDto;
+import de.symeda.sormas.api.action.ActionReferenceDto;
+import de.symeda.sormas.api.caze.CaseReferenceDto;
+import de.symeda.sormas.api.contact.ContactReferenceDto;
+import de.symeda.sormas.api.document.DocumentDto;
+import de.symeda.sormas.api.document.DocumentRelatedEntityType;
+import de.symeda.sormas.api.event.EventReferenceDto;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.backend.document.DocumentFacadeEjb;
+import de.symeda.sormas.backend.user.UserService;
+
+public class DocGenerationHelper {
```

Comment:
would be better to make this a stateless local bean so `userService` and `documentFacade` could be injected as EJB and no constructor would be needed

Reply:
It cannot be done due to the point of how CGLIB is working. They will always be null if it will not be called through an interface. Initially I did it in that way, but afterwards @EJB was not created and `null pointer exception` was raised.
![image](https://user-images.githubusercontent.com/6985315/148766274-4e328844-5615-48cc-a5ab-77605d9cfba6.png)